src/c4/ifpstream.hh

```diff
@@ -0,0 +1,54 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   c4/ifpstream.hh
+ * \author Mathew Cleveland
+ * \brief  Define class ifpstream
+ * \note   Copyright (C) 2018-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#ifndef c4_ifpstream_hh
+#define c4_ifpstream_hh
+
+#include <fstream>
+#include <sstream>
+#include <vector>
+
+#include "c4/config.h"
+
+namespace rtt_c4 {
+
+//============================================================================//
+/*!
+ * \class ifpstream
+ *
+ * Input stream for serializing input to all MPI ranks.
+ *
+ * This offloads all file IO to a single processor. The master processor reads
+ * all data from the file and broadcasts it to the MPI processors.
+ *
+ * \example c4/test/tstifpstream.cc
+ */
+//============================================================================//
+
+class ifpstream : public std::stringstream {
+public:
+  //! Constructor -- default to standard output mode (ASCII)
+  ifpstream(std::string const &filename,
+            ios_base::openmode const mode = ios_base::out);
+
```

Comment:
Please define the default constructor as `=deleted` or make a comment about why the base class default ctor is good enough.  If you delete the ctor, please use the "rule of 5" to also delete copy/move ctors and copy/move assignment operators.  If these are missing from the `ofpstream` class please add them or work with @keadyk to add them.

Reply:
deleted the default constructor in both ifpstream and ofpstream. I'm not going to specify the close on destruction because the default destructor should handle this when the object goes out of scope. Does that sound alright?